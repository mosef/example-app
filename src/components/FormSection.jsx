import React, { Component } from 'react';
import Form from './Form';

export default class FormSection extends Component {
  render() {
    return (
      <div>
        <h1 data-testid="page-title">New Form</h1>
        <Form />
      </div>
    );
  }
}
