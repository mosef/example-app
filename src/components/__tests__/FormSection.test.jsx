import React from 'react';
import { render, cleanup, fireEvent } from 'react-testing-library';
import FormSection from '../FormSection';

afterEach(cleanup);

test('<FormSection>', () => {
  const { debug, getByTestId, queryByTestId, container, getByText } = render(
    <FormSection />
  );
  expect(getByTestId('page-title').textContent).toBe('New Form');
  expect(queryByTestId('form')).toBeTruthy();
  debug();
  expect(container.firstChild).toMatchSnapshot();
});
