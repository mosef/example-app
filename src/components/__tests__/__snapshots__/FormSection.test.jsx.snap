// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormSection> 1`] = `
<div>
  <h1
    data-testid="page-title"
  >
    New Form
  </h1>
  <form
    data-testid="form"
  >
    <input
      type="text"
    />
    <button>
      Submit
    </button>
  </form>
</div>
`;
